db/names.sql

```diff
@@ -0,0 +1,504 @@
+--
+-- PostgreSQL database dump
+--
+
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SET check_function_bodies = false;
+SET client_min_messages = warning;
+
+SET search_path = public, pg_catalog;
+
+
+--
+-- Name: plpgsql; Type: EXTENSION; Schema: -; Owner: -
+--
+
+CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;
+
+
+--
+-- Name: EXTENSION plpgsql; Type: COMMENT; Schema: -; Owner: -
+--
+
+COMMENT ON EXTENSION plpgsql IS 'PL/pgSQL procedural language';
+
+--
+-- Name: instrumentnames; Type: TABLE; Schema: public; Owner: -; Tablespace: 
+--
+
+SET search_path = public, pg_catalog;
+
+CREATE FUNCTION f_concat_rd(array_type text, array_name text, site text, platform text, assembly text, instrument_name text) RETURNS text
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    IF assembly IS NOT NULL AND instrument_name IS NOT NULL THEN
+        RETURN concat(array_type, ' ', array_name, ' ', site, ' ', platform, ' - ', assembly, ' - ', instrument_name);
+    ELSIF assembly IS NOT NULL AND instrument_name IS NULL THEN
+        RETURN concat(array_type, ' ', array_name, ' ', site, ' ', platform, ' - ', assembly);
+    ELSE
+        RETURN concat(array_type, ' ', array_name, ' ', site, ' ', platform);
+    END IF;
+END
+$$;
+
+--
+-- Name: f_display_name(text); Type: FUNCTION; Schema: public; Owner: -
+--
+
+CREATE FUNCTION f_display_name(reference_designator text) RETURNS text
+    LANGUAGE plpgsql
+    AS $_$
+DECLARE
+    p_n platformnames%rowtype;
+    i_n instrumentnames%rowtype;
+    assy TEXT;
+    inst TEXT;
+    platform_text TEXT;
+    rd_len INT;
+BEGIN
+    rd_len := char_length(reference_designator);
+
+    IF NOT reference_designator ~ 'MOAS' THEN
```

Comment:
This is the majority of the reason we need a function in lieu of actual rows. The whole vocabulary spreadsheet has a lot of dynamic content that acts as a function of the reference designator.